File: ui/kotlin/src/main/kotlin/com/neo/ui/NeoApp.kt
package com.neo.ui

class NeoApp(private val config: AppConfig = AppConfig()) {
    private val state = AppState()
    val currentState: AppState get() = state

    fun initialize() {
        state.initialized = true
    }

    fun navigate(screen: String) {
        state.currentScreen = screen
    }

    fun shutdown() {
        state.initialized = false
    }
}

data class AppConfig(
    val title: String = "Neo AGI OS",
    val theme: String = "dark",
    val debugMode: Boolean = false,
)

class AppState {
    var initialized: Boolean = false
    var currentScreen: String = "home"
    var notifications: MutableList<String> = mutableListOf()

    fun addNotification(message: String) {
        notifications.add(message)
    }

    fun clearNotifications() {
        notifications.clear()
    }
}

File: ui/swift/Sources/NeoUI/NeoApp.swift
import Foundation

public class NeoApp {
    public struct Config {
        public let title: String
        public let theme: String
        public let debugMode: Bool

        public init(title: String = "Neo AGI OS", theme: String = "dark", debugMode: Bool = false) {
            self.title = title
            self.theme = theme
            self.debugMode = debugMode
        }
    }

    public struct State {
        public var initialized: Bool = false
        public var currentScreen: String = "home"
        public var notifications: [String] = []

        public init() {}
    }

    private let config: Config
    public private(set) var state: State

    public init(config: Config = Config()) {
        self.config = config
        self.state = State()
    }

    public func initialize() {
        state.initialized = true
    }

    public func navigate(to screen: String) {
        state.currentScreen = screen
    }

    public func notify(_ message: String) {
        state.notifications.append(message)
    }

    public func shutdown() {
        state.initialized = false
        state.notifications.removeAll()
    }
}
